src/lib/ivy/cache-compiler-host.ts

```diff
@@ -0,0 +1,166 @@
+  };
+
+  return {
+    ...compilerHost,
+
+    // ts specific
+    fileExists: (fileName: string) => {
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (cache.exists === undefined) {
+        cache.exists = compilerHost.fileExists.call(this, fileName);
+      }
+      return cache.exists;
+    },
+
+    getSourceFile: (fileName: string, languageVersion: ts.ScriptTarget) => {
+      addDependee(fileName);
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (!cache.sourceFile) {
+        cache.sourceFile = compilerHost.getSourceFile.call(this, fileName, languageVersion);
+      }
+      return cache.sourceFile;
+    },
+
+    writeFile: (
+      fileName: string,
+      data: string,
+      writeByteOrderMark: boolean,
+      onError?: (message: string) => void,
+      sourceFiles?: ReadonlyArray<ts.SourceFile>,
+    ) => {
+      if (fileName.endsWith('.d.ts')) {
+        sourceFiles.forEach(source => {
+          const cache = sourcesFileCache.getOrCreate(source.fileName);
+          if (!cache.declarationFileName) {
+            cache.declarationFileName = ensureUnixPath(fileName);
+          }
+        });
+      }
+
+      compilerHost.writeFile.call(this, fileName, data, writeByteOrderMark, onError, sourceFiles);
+    },
+
+    readFile: (fileName: string) => {
+      addDependee(fileName);
+
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (cache.content === undefined) {
+        cache.content = compilerHost.readFile.call(this, fileName);
+      }
+      return cache.content;
+    },
+
+    // ng specific
+    moduleNameToFileName: (moduleName: string, containingFile: string) => {
+      const { resolvedModule } = ts.resolveModuleName(
+        moduleName,
+        ensureUnixPath(containingFile),
+        compilerOptions,
+        compilerHost,
+        moduleResolutionCache,
+      );
+
+      return resolvedModule && resolvedModule.resolvedFileName;
+    },
+
+    resolveModuleNames: (moduleNames: string[], containingFile: string) => {
+      return moduleNames.map(moduleName => {
+        const { resolvedModule } = ts.resolveModuleName(
+          moduleName,
+          ensureUnixPath(containingFile),
+          compilerOptions,
+          compilerHost,
+          moduleResolutionCache,
+        );
+
+        return resolvedModule;
+      });
+    },
+
+    resourceNameToFileName: (resourceName: string, containingFilePath: string) => {
+      const resourcePath = path.resolve(path.dirname(containingFilePath), resourceName);
+      const containingNode = getNode(containingFilePath);
+      const resourceNode = getNode(resourcePath);
+      containingNode.dependsOn(resourceNode);
+
+      return resourcePath;
+    },
+
+    readResource: async (fileName: string) => {
+      addDependee(fileName);
+
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (cache.content === undefined) {
+        if (/(html|htm|svg)$/.test(path.extname(fileName))) {
```

Comment:
nit:

```suggestion
        if (/(?:html?|svg)$/.test(path.extname(fileName))) {
```

Reply:
This is also still left unchanged. Did some commit go missing maybe?

Reply:
I've figured it out, there's two compiler hosts! See my comment on the other one.